CMakeLists.txt

```diff
@@ -1,33 +1,47 @@
 cmake_minimum_required(VERSION 3.17)
-project(trian)
+project(trian
+	DESCRIPTION "inglor's Delaunay triangulation"
+	HOMEPAGE_URL "https://github.com/inglor/trian"
+	LANGUAGES CXX
+)
 
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
+#NIKOS:
+#       Setting compiler options by hand is not the CMake way but it is the only way I know for these options
+#       Even so, the "clean" way is per target, but this is enough for this project.
+if( CMAKE_CXX_COMPILER_ID MATCHES "GNU"  OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	message(STATUS "GCC/Clang detected, setting stricter than default warning flags")
+	list(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra -pedantic")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+	message(STATUS "MS Visual Studio, setting stricter than default warning flags")
+	list(APPEND CMAKE_CXX_FLAGS "/Wall /Za")
+endif()
```

Comment:
Add support for MacOs compiler?

Reply:
AFAIK [CMAKE_CXX_COMPILER_ID](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html) is compiler specific not platform specific, ie GNU and CLang should also work on OSX as is.